vllm/model_executor/layers/logits_processor.py

```diff
@@ -84,21 +84,29 @@ def _apply_logits_processors(
     logits: torch.Tensor,
     sampling_metadata: SamplingMetadata,
 ) -> torch.Tensor:
-    logits_row_idx = 0
+    logits_row_idx = -1
     found_logits_processors = False
     for seq_ids, sampling_params in sampling_metadata.seq_groups:
         logits_processors = sampling_params.logits_processors
         if logits_processors:
             found_logits_processors = True
             for seq_id in seq_ids:
+                # handle prompt_logprobs by skipping rows in logits added for
+                # the prompt tokens (prompt logprobs are not processed)
+                if (seq_id < sampling_metadata.num_prompts
+                        and sampling_params.prompt_logprobs is not None):
+                    logits_row_idx += sampling_metadata.prompt_lens[seq_id]
```

Comment:
```suggestion
                    logits_row_idx += sampling_metadata.prompt_lens[seq_id] - 1
```
and revert the other `logits_row_idx` changes? I think that would make things a bit clearer.